plugins/woocommerce/templates/single-product/product-image.php

```diff
@@ -42,9 +43,10 @@
 		if ( $post_thumbnail_id ) {
 			$html = wc_get_gallery_image_html( $post_thumbnail_id, true );
 		} else {
-			$html  = '<div class="woocommerce-product-gallery__image--placeholder">';
-			$html .= sprintf( '<img src="%s" alt="%s" class="wp-post-image" />', esc_url( wc_placeholder_img_src( 'woocommerce_single' ) ), esc_html__( 'Awaiting product image', 'woocommerce' ) );
-			$html .= '</div>';
+			$wrapper_classname = $has_images ? 'woocommerce-product-gallery__image woocommerce-product-gallery__image--placeholder' : 'woocommerce-product-gallery__image--placeholder';
```

Comment:
If there are variation images, I'm keeping both classes: `woocommerce-product-gallery__image` and `woocommerce-product-gallery__image--placeholder`. I think that's better for backwards-compatibility, in case somebody wrote some CSS based on  the `--placeholder` class. Also because until a variation is selected, the placeholder image is rendered, so I think it makes sense to keep the `--placeholder` class.